postgresql-11.5/src/bin/psql/commandbteq.c

```diff
@@ -208,11 +212,12 @@ exec_command(const char *cmd,
                    cmd);
     }
 
-    if (strcasecmp(cmd, "logon") == 0) {
+    if (strcasecmp(cmd, "logon") == 0 || strcasecmp(cmd, "logon;") == 0) {
```

Comment:
What if we provide `logon;;` or `logon ;;; ; ;`? Need to find a generic solution, not to cover a specific case.